### DISCUSSION TITLE
Can lazy load my own code but not npm packages

### DISCUSSION THREAD

**Discussion by momander** [quasarframework/quasar#12258] [General - Components / Directives / etc] [answered]
I read the page about lazy-loading and found it very helpful. When I do this in my code:

`const utils = await import(/* webpackChunkName: "utils" */'./utils.js');`

a separate utils.xxxxx.js bundle is created in dist/spa/js. Great!

But I can't get lazy loading of npm packages to work. For example, I use the library node-vibrant to find dominant colors in images. First I install it:

`npm install node-vibrant`

Then in my code I import it right before using it:

`const vibrant = await import(/* webpackChunkName: "vibrant" */ 'node-vibrant')`

Then I run `quasar build`. I expected a separate "vibrant" bundle in dist/spa/js, but it's not there. The node-vibrant package has been included in vendor.xxxxx.js instead. I get the same result if I skip webpackChunkName:

`const vibrant = await import('node-vibrant')`

Lazy loading npm packages worked in the old (pre-Quasar) version of my application. What am I doing wrong?

**Comment by yusufkandemir** [comment, marked as answer]
You just need to add `node-vibrant` to `quasar.conf.js > vendor > remove`, and it will not be included in the vendor chunk. Since you have specified `/* webpackChunkName: "vibrant" */` magic comment, it will be put into a new chunk called `vibrant` as you would expect.
See the docs for some explanation: https://quasar.dev/quasar-cli/quasar-conf-js#property-vendor

**Reply by momander** [reply]
Works great. Thank you for a quick and helpful answer!

I think the page about lazy loading should mention this, and point to https://quasar.dev/quasar-cli/quasar-conf-js#property-vendor. I tried creating a pull request for adding a tip section to https://quasar.dev/quasar-cli/lazy-loading#lazy-load-on-the-fly. Here is the markup for it:

```
::: tip
If you are importing an installed Node module (instead of your own code) and 
you want it to be in its own bundle instead of the app bundle, remember to 
adjust the [vendors section](https://quasar.dev/quasar-cli/quasar-conf-js#property-vendor)
of ````quasar.conf.js````.
:::
```

But the tip box does not render correctly in the online editor's preview. Should I create the pull request some other way?

**Reply by yusufkandemir** [reply]
That makes sense, it's a good idea.

You can think of tip/warning etc. boxes as special boxes we implemented, they don't normally exist within general markdown implementations. So, it's not rendered in GitHub editor or any other markdown editor. But, it will be rendered on docs correctly. You can read [Contributing to the Quasar Documentation](https://quasar.dev/contribution-guide/contribution-guide#contributing-to-the-quasar-documentation) on how to get it running on your machine.

However, I think this tip box is a bit vague, a real example would be much better, it might also give some imagination. So, if you can adjust your explanation, add some example code, add that to [Lazy-load on the fly](https://quasar.dev/quasar-cli/lazy-loading#lazy-load-on-the-fly) section and PR it, that would be great!

**Reply by momander** [reply]
Ah, got it. Yes, the tip box would be more helpful if it had an example. Will rewrite it and submit a PR!

**Reply by momander** [reply]
PR submitted.
